### DISCUSSION TITLE
[Windows] Can I custom installation directory?

### DISCUSSION THREAD

**Discussion by uqsme** [ppy/osu#20409] [Q&A]
it seems that lazer is automatically installed in my user directory (commonly in C drive), what if i want it to be in the D drive or somewhere else?

**Comment by peppy** [comment]
If you just want to move the user data, there's a "change folder" button towards the top of settings.

![osu! 2022-09-23 at 04 07 47](https://user-images.githubusercontent.com/191335/191890467-91a140fd-4a45-4a25-895d-41b06e815687.png)

**Reply by ghost** [reply]
but why do you force the installation folder as appdata/local? why I can't choose which location or drive I want to install the game? the same for user data location, which is in roaming folder by default.

so is there any way to set a custom folder for game and user data location?

**Reply by Cootz** [reply]
No, you can only change data folder location. There are some discussions about a custom installer for lazer, but idk if devs even plan to add it

**Reply by peppy** [reply]
The game is <200mb. It shouldn't matter where it is. But you can move it manually if you so wish.

**Reply by flowyun** [reply]
Haha infact it is a .zip file , just look it is a .exe . you can unzip to D:

**Reply by flowyun** [reply]
It is a self-extracting format file

**Reply by ghost** [reply]
> It shouldn't matter where it is. 

Thanks for your answer. Actually, I’m talking about why there is no ability to choose the installation directory like we have in stable installer/updater. The latter just picks appdata by default, but I can click and change it to another. 

I thought it’s something like “lazer’s legacy”, like for a development build, but things never change with time for some reason. ~Even no cutie window and some greetings text, just a logo and self-extraction.~

I mean the same for user data: the game uses “osu” folder in appdata/roaming for maps, skins and some caches. But, even if I change it to something else through the game settings, the abovementioned folder continues to exist and contains some files and cache/fonts folder which takes is about 150mb. Removing this folder obviously resets the game.

Well, I wrote too much, what a shame… But I would welcome that ability in the future. It would probably be even better to keep it in a single folder with subfolders, wouldn't it?
